mfr/extensions/tabular/libs/xlrd_tools.py

```diff
@@ -43,4 +43,7 @@ def xlsx_xlrd(fp):
         header = header_population(fields)
         sheets[sheet.name] = (header, data)
 
+    if not sheets:
```

Comment:
Sheets should be kept as an ordered dictionary as tab order is important in spreadsheets.

Also you might want to check with Erin if we should remove blank sheets or display them w/o any data (my preference would be the latter).